### PR TITLE
shared/filestore/localstore.go: remove unnecessary warning

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
I didn't get why this was there anyway, if `maxDepth` is `0`, we just stop recursion anyway. This is also not an actionable item from the administrator point of view right?

I mean this always warns when using local filebackend's list directory method.

```Go
func (b *LocalFileBackend) ListDirectory(path string) ([]string, error) {
	return appendRecursively(b.directory, path, 0)
}
```

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by Willyfrog — APPROVED** [review]
I don't recall the details, as this was a while ago, but basically to let the user know why it stopped.
Anyway, it is a very weird situation to happen, so 0/5 and approving

File: server/platform/shared/filestore/localstore.go [hunk 1]
```diff
@@ -211,7 +211,6 @@ func appendRecursively(basePath, path string, maxDepth int) ([]string, error) {
 		}
 		if dirEntry.IsDir() {
 			if maxDepth <= 0 {
-				mlog.Warn("Max Depth reached", mlog.String("path", entryPath))
```
Comment (agnivade) [inline server/platform/shared/filestore/localstore.go:214]: But the administrator should be aware that some files/directories will get skipped. This is important when running compliance export for example.

Maybe we can have a better error message: "Max depth reached, skipping any further directories under:"

**Review by agnivade — CHANGES_REQUESTED** [review]

**Comment by agnivade** [comment]
> I mean this always warns when using local filebackend's list directory method.

This is the real point. It's because we are using the same function for both. Suggest to have a separate non-recursive function for `ListDirectory`, or have a flag `recurse` and don't log the warning if that's set to false. 0/5

File: server/platform/shared/filestore/localstore.go (hunk 1 above)
Comment (isacikgoz) [reply server/platform/shared/filestore/localstore.go:214]: Hmm, but it will always print warning if you want to just list directories with 0 recursion.

**Comment by isacikgoz** [comment]
> > I mean this always warns when using local filebackend's list directory method.
> 
> This is the real point. It's because we are using the same function for both. Suggest to have a separate non-recursive function for `ListDirectory`, or have a flag `recurse` and don't log the warning if that's set to false. 0/5

@agnivade I think from the compliance point of view, we are not concerned if we don't hit the max depth right?

**Comment by agnivade** [comment]
Yes. So to confirm, what I am suggesting is:
1. No warning if ListDirectory gets called.
2. Warning with a better message when ListDirectoryRecursively gets called.
3. Bump up the limit to 50.

**Review by agnivade — APPROVED** [review]
It would be great if we could have a test that verifies that the warn log is no longer printed in the non-recursive case, but it's gonna be too complicated to mock everything. Looks good!

**Comment by isacikgoz** [comment]
@agnivade @Willyfrog since there is no actual behaviour change, I'm inclined to merge without QA, thought?

**Comment by agnivade** [comment]
Let's atleast run a e2e

/e2e-test

**Pull request merged**